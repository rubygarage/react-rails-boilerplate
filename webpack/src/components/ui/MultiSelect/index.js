import React, { Component } from 'react'
import classNames from 'classnames'
import ReactSelect from 'react-select'
import { Field } from 'redux-form'

class MultiSelect extends Component {
  field = ({ input, name, options, placeholder, onOpen, isLoading, classWrap, className, meta: { touched, error } }) => (
    <div className={classNames(classWrap, { 'has-error': (touched && error) })}>
      <ReactSelect
        {...input}
        multi
        onOpen={onOpen}
        isLoading={isLoading}
        value={input.value}
        name={name}
        options={options}
        placeholder={placeholder}
        className={className}
        onChange={(values) => input.onChange(values.map((value) => value.value))}
        onBlur={() => input.onBlur(input.value)}
      />
      { touched && error && <p className="error">{error}</p> }
    </div>
  )

  render() {
    return <Field {...this.props} component={this.field} />
  }
}

export default MultiSelect
