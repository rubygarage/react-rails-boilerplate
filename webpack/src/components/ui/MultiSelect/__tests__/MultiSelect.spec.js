import React from 'react'
import { shallow } from 'enzyme'
import MultiSelect from 'components/ui/MultiSelect'

describe('<MultiSelect />', () => {
  const defaultProps = {
    options: [{ value: 1, lable: 'Mood' }]
  }

  it('renders with empty props', () => {
    const component = shallow(<MultiSelect />)

    expect(component).toMatchSnapshot()
  })

  it('renders with props', () => {
    const component = shallow(<MultiSelect {...defaultProps} />)

    expect(component).toMatchSnapshot()
  })
})
