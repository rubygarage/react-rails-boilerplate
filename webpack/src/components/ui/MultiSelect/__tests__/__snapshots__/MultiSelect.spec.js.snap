// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiSelect /> renders with empty props 1`] = `
<Field
  component={[Function]}
/>
`;

exports[`<MultiSelect /> renders with props 1`] = `
<Field
  component={[Function]}
  options={
    Array [
      Object {
        "lable": "Mood",
        "value": 1,
      },
    ]
  }
/>
`;
